include/nbl/builtin/glsl/format/encode.glsl

```diff
@@ -82,4 +82,51 @@ uint nbl_glsl_encodeRGB10A2_SNORM(in vec4 col)
 	return quantized.r|quantized.g|quantized.b|quantized.a;
 }
 
+// TODO: break it down into uint nbl_glsl_encode_ufloat_exponent(in float _f32) and nbl_glsl_encode_ufloat_mantissa(in float _f32, in uint mantissaBits, in bool leadingOne)
+uint nbl_glsl_encode_ufloat(in float _f32, in uint mantissaBits)
+{
+	uint minSinglePrecisionVal = floatBitsToUint(nbl_glsl_R11G11B10_MIN);
+	uint maxSinglePrecisionVal = floatBitsToUint(nbl_glsl_R11G11B10_MAX);
+
+	if (_f32 < uintBitsToFloat(maxSinglePrecisionVal))
+	{
+		if (_f32 < uintBitsToFloat(minSinglePrecisionVal))
```

Comment:
see https://github.com/Devsh-Graphics-Programming/Nabla/pull/343/files#r906730818

also why are you `reinterpret_cast`ing back and forth?

Reply:
my bad I did that XD